### PR TITLE
Fix some unittests with locale de_DE.UTF-8

### PR DESCRIPTION
**Pull request by stweil**
The unittest failed with LANG=de_DE.UTF-8:

    $ unittest/apiexample_test
    Running main() from ../../../../unittest/../googletest/googletest/src/gtest_main.cc
    [==========] Running 4 tests from 2 test suites.
    [----------] Global test environment set-up.
    [----------] 1 test from EuroText
    [ RUN      ] EuroText.FastLatinOCR
    contains_unichar_id(unichar_id):Error:Assert failed:in file ../../../../../src/ccutil/unicharset.h, line 874

Signed-off-by: Stefan Weil <sw@weilnetz.de>

### CODE REVIEW

**Comment by zdenop** [comment]
This pull request **introduces 1 alert** when merging 0dcc889e8d9b029026d3aa6a6d6b1f1d1b890c6a into 4b397c70cc7d2aef2e50cdb9581b7e10f789ec3d - [view on LGTM.com](https://lgtm.com/projects/g/tesseract-ocr/tesseract/rev/pr-ca8ba7562e7b65a1d73cc6a39cdfe30f45187773)

**new alerts:**

* 1 for FIXME comment

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by stweil** [comment]
I added the FIXME because there follows a `snprintf` statement which formats double or float values. The result will depend on the locale settings, so that needs a fix, too. But first I have to find a test case which triggers that code.

File: src/ccutil/unicharset.cpp [hunk 1]
```diff
@@ -706,6 +709,7 @@ bool UNICHARSET::save_to_string(STRING *str) const {
               this->get_script_from_script_id(this->get_script(id)),
               this->get_other_case(id));
     } else {
+      // FIXME
```
Comment (amitdo) [inline src/ccutil/unicharset.cpp]: You should clarify what is needed to be fixed here

File: src/ccutil/unicharset.cpp [hunk 2]
```diff
@@ -815,41 +819,64 @@ bool UNICHARSET::load_via_fgets(
     float advance = 0.0f;
     float advance_sd = 0.0f;
     // TODO(eger): check that this default it ok
-    // after enabling BiDi iterator for Arabic+Cube.
+    // after enabling BiDi iterator for Arabic.
```
Comment (amitdo) [inline src/ccutil/unicharset.cpp]: Maybe it should be Arabic+LSTM ?

Comment (stweil) [reply src/ccutil/unicharset.cpp]: Maybe. The more important point is that this TODO is either still open, or the comment should be removed.

File: src/ccutil/unicharset.cpp (hunk 1 above)
Comment (amitdo) [reply src/ccutil/unicharset.cpp]: ... by adding a comment after the FIXME

Comment (stweil) [reply src/ccutil/unicharset.cpp]: I should have done that, yes. But instead of fixing the FIXME comment, I prefer to fix the code, hopefully today.

Comment (amitdo) [reply src/ccutil/unicharset.cpp]: OK :-)

**Comment by zdenop** [comment]
@stweil : is this PR ready for merge or do you plan to add something to it?

**Comment by zdenop** [comment]
This pull request **introduces 1 alert** when merging 36ed6da3499c93c2d04de29ee2f02f6d9975a1fe into 4b397c70cc7d2aef2e50cdb9581b7e10f789ec3d - [view on LGTM.com](https://lgtm.com/projects/g/tesseract-ocr/tesseract/rev/pr-05855c547ab62cd9493a344ca3a3fff2830f2bd1)

**new alerts:**

* 1 for FIXME comment

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/ccutil/unicharset.cpp (hunk 1 above)
Comment (stweil) [reply src/ccutil/unicharset.cpp]: `save_to_string` is fixed now, too.

**Comment by stweil** [comment]
> is this PR ready for merge or do you plan to add something to it?

Now I think that it is ready for merging. The last commit is the one we were waiting for: it removes the assertions which check the locale.

**Comment by zdenop** [comment]
Thanks

**Pull request merged**